examples/blink/CMakeLists.txt

```diff
@@ -2,6 +2,17 @@ cmake_minimum_required(VERSION 3.12)
 
 set(ROOT_DIR ${CMAKE_CURRENT_LIST_DIR}/../../)
 set(CMAKE_TOOLCHAIN_FILE ${ROOT_DIR}/toolchain-rv32.cmake)
+# ESP Target MCU. Supported: esp32c3, esp32h2
+list(APPEND supported_mcu esp32c3 esp32h2)
+
+if(target IN_LIST supported_mcu)
+    message(STATUS "Configure project for ${target} ... ")
+else ()
+    message(FATAL_ERROR "Supported target ${supported_mcu}")
+endif()
+string(TOUPPER ${target} target_uppercase)
+set(${target_uppercase} 1)
+configure_file(${ROOT_DIR}/config.h.in ${ROOT_DIR}/config.h)
```

Comment:
In CMake it is considered a good practice to keep all the generated files in the build directory. This allows having builds e.g. for different targets side by side. Could you move this generated file to the build directory?

Reply:
Yes. agreed. 

But in this case I need to include the `${project_dir}/example/example_name/build` folder to `${project_dir}/common/` depending on the example which is building ... which seems little bit odd, no?

I'll try to do something with that.

Reply:
The normal practice in CMake is to create generated files inside ${CMAKE_CURRENT_BINARY_DIR}.

For example, create this file in ${CMAKE_CURRENT_BINARY_DIR}/config/config.h and then add ${CMAKE_CURRENT_BINARY_DIR}/config using target_include_directories.